crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -0,0 +1,948 @@
+            Operation::fneg => todo!(),
+            Operation::fdiv => todo!(),
+            Operation::div => todo!(),
+            Operation::eq => BigUint::from(1_u32),
+            Operation::ne => BigUint::from(1_u32),
+            Operation::ugt => BigUint::from(1_u32),
+            Operation::uge => BigUint::from(1_u32),
+            Operation::ult => BigUint::from(1_u32),
+            Operation::ule => BigUint::from(1_u32),
+            Operation::sgt => BigUint::from(1_u32),
+            Operation::sge => BigUint::from(1_u32),
+            Operation::slt => BigUint::from(1_u32),
+            Operation::sle => BigUint::from(1_u32),
+            Operation::lt => BigUint::from(1_u32),
+            Operation::gt => BigUint::from(1_u32),
+            Operation::lte => BigUint::from(1_u32),
+            Operation::gte => BigUint::from(1_u32),
+            Operation::and => BigUint::from(1_u32),
+            Operation::not => BigUint::from(1_u32),
+            Operation::or => BigUint::from(1_u32),
+            Operation::xor => BigUint::from(1_u32),
+            //'a cast b' means we cast b into a: (a) b
+            //we assume that eventual truncate has been done so rhs_max must fit into type of a.
+            Operation::cast => rhs_max,
+            Operation::trunc => BigUint::min(
+                lhs_max,
+                BigUint::from(2_u32).pow(rhs_max.try_into().unwrap()) - BigUint::from(1_u32),
+            ),
+            //'a = b': a and b must be of same type.
+            Operation::ass => rhs_max,
+            Operation::nop | Operation::jne | Operation::jeq | Operation::jmp => todo!(),
+            Operation::phi => BigUint::max(lhs_max, rhs_max), //TODO operands are in phi_arguments, not lhs/rhs!!
+            Operation::eq_gate => BigUint::min(lhs_max, rhs_max),
+        }
+    }
+
+    // pub fn get_const_value(c: &Constant) -> (u32, u32) {
+    //     //....todo...only u32 for now... should also provide the sign
+    //     match c.value_type {
+    //         ObjectType::boolean => (if c.value.is_zero() { 0 } else { 1 }, 1),
+    //         ObjectType::native_field => todo!(), //to field(value), field:prime
+    //         ObjectType::signed(b) | ObjectType::unsigned(b) => {
+    //             (c.value.clone().try_into().unwrap(), b)
+    //         }
+    //         _ => todo!(),
+    //     }
+    // }
+
+    pub fn get_const_value2(c: FieldElement, ctype: ObjectType) -> (u32, u32) {
+        //....todo...only u32 for now... should also provide the sign
+        match ctype {
+            ObjectType::boolean => (if c.is_zero() { 0 } else { 1 }, 1),
+            ObjectType::native_field => (c.to_u128().try_into().unwrap(), 32), //TODO! //to field(value), field:prime
+            ObjectType::signed(b) | ObjectType::unsigned(b) => (c.to_u128().try_into().unwrap(), b),
+            _ => todo!(),
+        }
+    }
+
+    // pub fn to_const(value: FieldElement, obj_type: ObjectType) -> NodeObj {
```

Comment:
dead code?

Reply:
yes, it has been removed as per Jake's comment